project_euler/problem_01/sol8.py

```diff
@@ -0,0 +1,33 @@
+"""
+Problem Statement:
+If we list all the natural numbers below 10 that are multiples of 3 or 5,
+we get 3,5,6 and 9. The sum of these multiples is 23.
+Find the sum of all the multiples of 3 or 5 below N.
+"""
+
+
+def solution(n: int) -> int:
+    """Returns the sum of all the multiples of 3 or 5 below n.
+    What unique about this solution?
+    Although this solution is slower than previous solutions,
+    but it is more elegant and clear.
+    It uses Python features to do some logic,
+    like using range's step to check for only multiples of 3 and 5
+    and use set's union to remove duplication
+
+    >>> solution(3)
+    0
+    >>> solution(4)
+    3
+    >>> solution(10)
+    23
```

Comment:
```suggestion
    >>> [solution(i) for i in range(11)]
    [ ... ]
```

Reply:
Also what happens when we `solution(-1)` or `solution(3.3)`

Reply:
The problem statement has a constraint of natural numbers less than 100

Reply:
Then the code needs to raise ValueError n is not a natural number < 100.